### PR TITLE
Update InfluxDBHandler for post InfluxDB 0.9

### PR DESCRIPTION
**Pull request by cj-dimaggio**
- Fix for ticket #297 where InfluxDBHandler was formatting metrics based
  on an older version of influxdb-python
- Maintains support for InfluxDB version 0.8 by way of an additional
  'influxdb_version' attribute in the config.
- Reformats measurement schema to be more useful for InfluxDB 0.9's removal
  of merge and joins.
- Adds several test cases for the handler.

### CODE REVIEW

**Comment by shortdudey123** [comment]
Can you update the doc as well and then sqaush that commit in?

**Comment by shortdudey123** [comment]
Can you remove the commits that are not yours and not related to this PR? (they are already in master)

File: docs/handlers/GraphiteHandler.md [hunk 1]
```diff
@@ -34,3 +34,4 @@ scope_id | 0 | IPv6 Scope ID | int
 server_error_interval | 120 | How frequently to send repeated server errors | int
 timeout | 15 |  | int
 trim_backlog_multiplier | 4 | Trim down how many batches | int
+reconnect_interval | 0 | How often (in seconds) to reconnect to graphite. Default (0) is never | int
```
Comment (shortdudey123) [inline docs/handlers/GraphiteHandler.md]: alphabetize

**Comment by Ssawa** [comment]
Sorry about that, my git history got pretty messy as I was developing. Should be cleaned up

**Comment by shortdudey123** [comment]
@Ssawa can you rebase on master again to fix the conflict?

**Comment by shortdudey123** [comment]
Will also solve https://github.com/python-diamond/Diamond/issues/137

**Comment by szibis** [comment]
@Ssawa I just use your version from this PR and add more features.

https://github.com/szibis/diamond-influxdb

If anyone thinks this is helpful we can makes this as official PR in Diamond to finally have a working handler with latest Influxdb versions.

I use this version to migrate (replicate traffic with proper format) from graphite to Influxdb and I will update it to work as efficient as I can.

**Comment by Ssawa** [comment]
Hey folks, sorry for letting this fall off my radar for so long. After thoroughly fucking my git history it should be fixed now and all the changes rolled up into one commit and rebased with the current master so it should hopefully be good to go.

Also @szibis repo looks pretty sweet and I've personally not been keeping up with Diamond for awhile so I'm all for going with his changes

**Comment by szibis** [comment]
If this can be merged @shortdudey123 I will do my PR based on this changes or you prefer other options?

**Comment by NotSqrt** [comment]
I tested this pull request successfully with a 1.2 server.

**Comment by adimania** [comment]
I tested this pull request successfully with a 1.3 influxdb server. I would request you to merge this with master.

**Comment by akhiltak** [comment]
Hi guys! Any ETA on when this change will be merged to master?

**Comment by eshelman** [comment]
Another datapoint - I'm happily running this with `influxdb-1.3.5-1.x86_64` and hoping it'll be merged soon. Thanks for this awesome piece of software!

**Comment by mcaubet** [comment]
Test with influxdb-1.6.3, no problems. Hope it will be merged some day.

**Pull request merged**